src/main/scala/cromwell/engine/backend/jes/JesConfiguration.scala

```diff
@@ -0,0 +1,22 @@
+package cromwell.engine.backend.jes
+
+/**
+ * Trait for JesConfiguration
+ */
+trait JesConfiguration {
```

Comment:
So the whole idea behind this pattern with trait + object + "Singleton" trait was to address 2 things:

- Provide any class that needs it with access to the Jes configuration. The main constraint is to provide all classes/traits with the SAME instance of `JesAttributes` as we don't want to reparse application.conf and rebuild a `JesAttributes` instance every time.

- Be able to test methods that depends on the configuration traits with a custom configuration defined in the test class.

The trait defines what is needed (without implementing it)
The object instantiate and holds an instance of this thing (that can be used application-wide).
The "Singleton" trait implements the first trait, using the instance in the object. So this trait can be mixed anywhere but will still provide the same instance.

However if you want to test a behavior that requires a specific configuration, you can define another trait and use this as the configuration by mixing it when instantiating the class you want to test (see JesBackendSpec).

It's exactly the same thing with JesAuthentication.